### ISSUE TITLE
java.lang.ClassCastException: com.phrase.android.sdk.inject.PhraseContextWrapper cannot be cast to android.app.ContextImpl

### ISSUE THREAD

**Issue by mzsanford** [phrase/phrase-android#25]
I'm testing Phrase integration and my app has a `BroadcastReceiver` registered in the manifest.xml like so:

```
       <receiver android:name="com.redacted.AppUpdatedReceiver">
            <intent-filter>
                <action android:name="android.intent.action.MY_PACKAGE_REPLACED" />
            </intent-filter>
        </receiver>
```

And the class itself is a simple subclass of `BroadcastReceiver` implementing `onReceive`. If I comment this out of the manifest.xml everything works fine. When I have this enabled my app crashes soon after launch with:

```
    java.lang.RuntimeException: Unable to instantiate receiver com.redacted.AppUpdatedReceiver: java.lang.ClassCastException: com.phrase.android.sdk.inject.PhraseContextWrapper cannot be cast to android.app.ContextImpl
        at android.app.ActivityThread.handleReceiver(ActivityThread.java:4012)
        at android.app.ActivityThread.access$1400(ActivityThread.java:237)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1924)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7660)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
     Caused by: java.lang.ClassCastException: com.phrase.android.sdk.inject.PhraseContextWrapper cannot be cast to android.app.ContextImpl
        at android.app.ActivityThread.handleReceiver(ActivityThread.java:3996)
        at android.app.ActivityThread.access$1400(ActivityThread.java:237) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1924) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:223) 
        at android.app.ActivityThread.main(ActivityThread.java:7660) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
```

Device: Google Pixel 5, Android 11

**Comment by theSoenke** [comment]
Hi @mzsanford, the Android SDK is expecting an exact instance of `ContextImpl` in some places, which are causing this issue. For now, I would recommend removing `Phrase.wrapApplicationContext` from your `Application` class. We'll also likely release a new version soon which will remove this method.

**Comment by erdincozden** [comment]
Some problem : Unable to instantiate receiver com.google.firebase.iid.FirebaseInstanceIdReceiver: java.lang.ClassCastException: com.phrase.android.sdk.inject.PhraseContextWrapper cannot be cast to android.app.ContextImpl

**Comment by theSoenke** [comment]
Hey, thanks for your patience. We have now found a solution to be able to wrap the application context in an `Application` class. The fix is released in version 3.0.2. This should resolve the cast exception.

**Comment by jerepious** [comment]
Hello,

We have the same issue that occurs on the latest version of Android SDK `3.5.0` , I noticed a crash the first time of running the app with the SDK integrated on `release` Build variants.
 : java.lang.RuntimeException: Unable to instantiate receiver
                                                                                                    com.google.firebase.iid.FirebaseInstanceIdReceiver: java.lang.ClassCastException: com.phrase.android.sdk.inject.PhraseContextWrapper
                                                                                                    cannot be cast to android.app.ContextImpl

**Comment by drachim-dev** [comment]
Same issue here:

```
android.app.ActivityThread.handleReceiver
ActivityThread.java, line 4413
java.lang.ClassCastException: com.phrase.android.sdk.inject.PhraseContextWrapper cannot be cast to android.app.ContextImpl
```

**Comment by theSoenke** [comment]
Hi, this turned out to be a regression and has should be resolved in version 3.7.1 https://github.com/phrase/phrase-android/releases/tag/3.7.1
